WordPress/gutenberg
# Load variation font faces with theme relative urls using backend provided full urls (_links)

## What?
Use backend provided full urls to load font assets with theme relative urls.

## Why?
To load font faces dynamically in the browser, for example, when they change as a result of applying a theme style variation.
Fixes: https://github.com/WordPress/gutenberg/issues/59965

## How?
Uses the links provided by the global styles rest API to load the font assets using the full URLs instead of the theme-relative ones that don't work on the client.


## Testing Instructions
Try to load a style variation and check that the fonts referenced in those style variations load as expected. See the screencast below as example.

**I created a theme to test this quickly and visually:**
[font-variants.zip](https://github.com/user-attachments/files/17311496/font-variants.zip)

| Before |  After |
| --- | --- |
| The fonts defined in the variants are not loaded, so the fallback system fonts are rendered by the browser. | The fonts are loaded so the expected font assets are rendered. |



### Before:
[Screencast from 09-10-24 13:25:05.webm](https://github.com/user-attachments/assets/3ec7d144-8654-4228-a7d2-b297c901a349)

### After:
[Screencast from 09-10-24 13:23:40.webm](https://github.com/user-attachments/assets/7fa59bce-5360-477d-bde5-09368d02f6ef)



## Screenshots or screencast 
Testing with TT4. It is the same as with the custom  theme I created but the typographies are not so different visually so the changes are harder to see.




https://github.com/user-attachments/assets/adb2c5c1-bf84-40da-933a-e6a9513936b4
